source/actions3/AMActionRunner3.h

```diff
@@ -83,6 +83,10 @@ class AMActionRunner3 : public QObject
 	void addActionToQueue(AMAction3* action) { insertActionInQueue(action, -1); }
 	/// Insert an action at a position \c index in the queue.  If \c index is -1 or > queuedActionCount(), will append to the end of the queue. If \c index is 0, will insert at the beginning of the queue. This class takes ownership of the \c action and will log and delete it after running it.
 	void insertActionInQueue(AMAction3* action, int index);
+	/// Insert an action at a position \c index in the queue.  If \c index is -1 or > queuedActionCount(), will append to the end of the queue. If \c index is 0, will insert at the beginning of the queue. This class takes ownership of the \c action and will log and delete it after running it.
+	bool InsertActionToQueue(AMAction3 *action, int index);
```

Comment:
the name is slightly different. The original function called `insertAction**In**Queue`. The added function has the common codes to do the real "insert" into queue, so I named it `insertActionToQueue`. It should be protected.